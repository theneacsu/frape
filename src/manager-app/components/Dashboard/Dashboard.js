import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import getVisibleTodos from '../../../selectors/todo-app/todos'
import styles from './Dashboard.module.css'

export const Dashboard = props => (
  <div className={styles.div}>
    <h1 className={styles.h1}>{props.userInfo.displayName}</h1>
    <img
      src={props.userInfo.photoURL}
      alt="User profile phote"
      className={styles.img}
    />
    <p className={styles.p}>
      You have{' '}
      <Link to="/todos" className={styles.a}>
        {props.todosLeft} todo(s)
      </Link>{' '}
      left
    </p>
    <p className={styles.p}>
      Read your{' '}
      <Link to="/notes" className={styles.a}>
        notes
      </Link>
    </p>
  </div>
)

const mapStateToProps = state => ({
  userInfo: state.auth,
  todosLeft: getVisibleTodos(state.todos, state.todosFilters).filter(
    todo => !todo.completed
  ).length
})

export default connect(mapStateToProps)(Dashboard)
