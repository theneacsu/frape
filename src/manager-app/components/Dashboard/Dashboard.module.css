.div {
  background-color: #f2fdfb;
  text-align: center;
  flex: 1;
}

.h1 {
  margin: 0;
  font-size: 2.5rem;
  padding: 3rem 1.6rem;
}

.p {
  font-size: 2rem;
}

.a {
  text-decoration: none;
  color: #4cb69f;
  font-weight: bold;
  font-size: 2rem;
}

.img {
  width: 20rem;
  height: 20rem;
  border-radius: 50%;
  display: block;
  margin: 0 auto;
}
