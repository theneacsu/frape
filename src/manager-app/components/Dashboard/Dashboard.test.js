import React from 'react'
import { shallow } from 'enzyme'
import { Dashboard } from './Dashboard'

test('Should render correctly the Dashboard component', () => {
  const user = {
    uid: '123abc',
    photoURL:
      'https://lh4.googleusercontent.com/-3HA5dfxwrYU/AAAAAAAAAAI/AAAAAAAAEJ4/q6fYoZQpeEY/photo.jpg',
    displayName: 'Bradley Cooper',
    email: 'test@test.com'
  }
  const wrapper = shallow(<Dashboard userInfo={user} todosLeft={3} />)
  expect(wrapper).toMatchSnapshot()
})
