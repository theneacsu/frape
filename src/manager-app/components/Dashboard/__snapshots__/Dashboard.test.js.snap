// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly the Dashboard component 1`] = `
<div
  className="div"
>
  <h1
    className="h1"
  >
    Bradley Cooper
  </h1>
  <img
    alt="User profile phote"
    className="img"
    src="https://lh4.googleusercontent.com/-3HA5dfxwrYU/AAAAAAAAAAI/AAAAAAAAEJ4/q6fYoZQpeEY/photo.jpg"
  />
  <p
    className="p"
  >
    You have
     
    <Link
      className="a"
      replace={false}
      to="/todos"
    >
      3
       todo(s)
    </Link>
     
    left
  </p>
  <p
    className="p"
  >
    Read your
     
    <Link
      className="a"
      replace={false}
      to="/notes"
    >
      notes
    </Link>
  </p>
</div>
`;
